### DISCUSSION TITLE
"ld returned 1 exit status" hatası alıyorum

### DISCUSSION THREAD

**Discussion by aykhanshirinzade** [19/classroom#65] [C]
Selamun Aleykum, hocam. Eksik de olsa kodu yazdım, 9'cu koşulu yapmağa çalışırken hata alıyorum;
```
/usr/bin/ld: /tmp/ccl5bKDi.o: in function `main':
odev1.c:(.text+0x20c): undefined reference to `log2'
collect2: error: ld returned 1 exit status
```
Araştırdım, ama işeyarar bir şey pek bulamadım. Github'a eklememe sebebimde derlenirken hata vermesi.

**Comment by roktas** [comment]
Math kitaplığından `log2` kullanmışsın, fakat kod (Github tarafında) `-lm` anahtarı ile derlenmediğinden bu hatayı alıyorsun.  Çözüm?  Bu ödevde matematiksel bir şey kullanman gerekmiyor.  `log2` fonksiyonunu kullanmadan bir çözüm bul.

**Reply by aykhanshirinzade** [reply]
Hocam, 9'cu koşul enptropi'yle alakalı değil mi?
`entropy_formula = log2(aviable_characters**length_of_password)`

Sizin yazdığınız gibi ($6.55 L$) aratınca hiçbir şey bulamadım. İpucu verebilir misiniz? :)

**Comment by roktas** [comment]
Entropi formülünü verdim ya açıklamada.  Dikkatli oku açıklamayı.
